### ISSUE TITLE
Make opentelemetry-instrumentation-logging an optional dependency

### ISSUE THREAD

**Issue by asisloustau** [logzio/logzio-python-handler#81]
Currently, logzio-python-handler has opentelemetry-instrumentation-logging as a required dependency - see: https://github.com/logzio/logzio-python-handler/blob/master/setup.py#L18

This `opentelemetry-instrumentation-logging` depends on `open-telemetry-instrumentation`. This package imports pkg_resources, which poses a few issues:
- Package discovery and resource access using `pkg_resources` is deprecated, and `importlib` should be used instead - see https://setuptools.pypa.io/en/latest/pkg_resources.html 
    - Because of this, if an app is shipped without setuptools, importing `LoggingInstrumentor` (https://github.com/logzio/logzio-python-handler/blob/master/logzio/handler.py#L11) breaks the app.
    - If an app is shipped with the latest `setuptools`, a `DeprecationWarning` is raised:

```
.venv/lib/python3.10/site-packages/opentelemetry/instrumentation/dependencies.py:4: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
  from pkg_resources import (
.venv/lib/python3.10/site-packages/pkg_resources/__init__.py:2871: DeprecationWarning: Deprecated call to `pkg_resources.declare_namespace('google')`.
Implementing implicit namespace packages (as specified in PEP 420) is preferred to `pkg_resources.declare_namespace`. See https://setuptools.pypa.io/en/latest/references/keywords.html#keyword-namespace-packages
  declare_namespace(pkg)
```
- The README for `opentelemetry-python-contrib` (which is `opentelemetry-instrumentation`, the installed dependency) states that "these libraries are currently in beta, and shouldn't generally be used in production environments" - see https://github.com/open-telemetry/opentelemetry-python-contrib/tree/main#installation.

Given the points described above, I'd like to suggest making opentelemetry-instrumentation-logging an optional dependency.
This could be accomplished by:
1. Updating `setup.py` to add an `extras_require` as follows:
```python
setup(
    ...
    extras_require = {
        "opentelemetry":  ["opentelemetry-instrumentation-logging"]
    }
)
```
This way, if users would like to use `add_context=True`, they will have to `pip install 'logzio-python-handler[opentelemetry]'`.
2. Importing [LoggingInstrumentor](https://github.com/logzio/logzio-python-handler/blob/master/logzio/handler.py#L11) only if `add_context` is `True` --> see https://github.com/logzio/logzio-python-handler/blob/master/logzio/handler.py#L33
(a try/catch with a user-friendly message could be added with the `ImportError`.)

This won't necessarily solve the opentelemetry issues, but it would give the user an option to:
1. Not have to add unneeded dependencies if opentelemetry is not used.
2. Not have to package dependencies that may not be production-ready.

Please let me know if you agree with these changes or if you have any feedback, and I will work on them if needed.

Thank you!

**Comment by ralongit** [comment]
Hello @asisloustau ,

Thank you for the feedback you have provided.
We have implemented these changes as part of version 4.1.0.

**Comment by asisloustau** [comment]
Thank you!
